### ISSUE TITLE
Dependencies only show one issue for each other repo

### ISSUE THREAD

**Issue by colejohnson66** [go-gitea/gitea#27722]
### Description

The "blocks/dependencies" area of issues only shows the three oldest additions to the list, even if there are more. The issue log does show there being more than three "new dependencies" being added. It appears like the code naively adds a `LIMIT 3` to the database call with no regard for the real count.

The entire list should be viewable, and it should be sorted differently than the order of insertion in the database.

### Gitea Version

1.20.5

### Can you reproduce the bug on the Gitea demo site?

Untested

### Screenshots

This screenshot shows only three issues as being blocked by the currently viewed one. The current issue actually blocks about a half-dozen. As above, these are the first three additions to the list, and appear in that order.

![Screenshot showing the above description](https://github.com/go-gitea/gitea/assets/11381599/a6871820-002f-464a-81ed-e3690949b5c5)

### How are you running Gitea?

N/A

### Database

MySQL/MariaDB

**Comment by colejohnson66** [comment]
I can't easily identify why this occurs. The [template](https://github.com/go-gitea/gitea/blob/b39bb958cc44511c3590e832f77cebdf70616b8c/templates/repo/issue/view_content/sidebar.tmpl#L429-L447) as well as [the code to retrieve them](https://github.com/go-gitea/gitea/blob/b39bb958cc44511c3590e832f77cebdf70616b8c/models/issues/issue.go#L662-L677) do not place a limit on the list's length.

**Comment by lng2020** [comment]
this PR https://github.com/go-gitea/gitea/pull/17935 has several bugs and typos. It really should add some tests to ensure its behavior.

**Comment by colejohnson66** [comment]
Looking at your new PR and the linked PR, I can indeed confirm that the issue (no pun intended) is that only the first added issue for each other repository is shown. And they are shown in order of repository ID.

So, for my screenshot, #&zwj;15 was from repo ID 2, #&zwj;856 from ID 3, and #&zwj;23 from ID 4. In addition, issue #&zwj;23; for repository ID 4 was actually the first to be added to the list of dependencies, but shows up last.

**Comment by lng2020** [comment]
That make much more sense. Your screenshot is strictly followed the old code logic.

**Comment by KazzmanK** [comment]
There is much to add to the Dependency list UI issues.
on try.gitea.io I'm unable to reproduce my problems
on our 1.20.5 it is shows Not found, or, when I type, it shows strange results in dropdown.

![image](https://github.com/go-gitea/gitea/assets/13328513/11358197-8365-48f6-91fb-84bf9a90279e)


Is there something fixed in upcoming releases ?

**Comment by KazzmanK** [comment]
It is still rising questions on 1.21
![image](https://github.com/go-gitea/gitea/assets/13328513/7939a193-5d21-45ab-9e29-970408e967c4)
